packages/marketplace/src/components/pages/admin-dev-management/admin-dev-management.tsx

```diff
@@ -42,16 +43,37 @@ export interface AdminDevManagementMappedProps {
 
 export type AdminDevManagementProps = AdminDevManagementMappedActions & AdminDevManagementMappedProps
 
-export const AdminDevManagement: React.FC<AdminDevManagementProps> = ({
-  adminDevManagementState,
-  filterValues,
-  onPageChange,
-  onSearch,
-  fetchData,
-}) => {
+export const buildFilterValues = (queryParams: URLSearchParams): AdminDevManagementFilterFormValues => {
+  const name = queryParams.get('name') || ''
+  const company = queryParams.get('company') || ''
+  const registeredFrom = queryParams.get('registeredFrom') || ''
+  const registeredTo = queryParams.get('registeredTo') || ''
+  return { name, company, registeredFrom, registeredTo } as AdminDevManagementFilterFormValues
+}
+
+export const handleFetchData = (dispatch: Dispatch) => (requestData: AdminDevManagementRequestDataValues) => {
+  dispatch(adminDevManagementRequestData(requestData))
+}
+
+export const AdminDevManagement: React.FC = () => {
+  const dispatch = useDispatch()
+
+  const history = useHistory()
+  const location = useLocation()
+  const { page = 1 } = useParams()
+
+  const fetchData = handleFetchData(dispatch)
+
+  const queryParams = new URLSearchParams(location.search)
```

Comment:
why don't we reuse this function to get queryParams
`const { page = 1 } = useParams()`

Reply:
> why don't we reuse this function to get queryParams
> `const { page = 1 } = useParams()`

Hi @tanphamhaiduong, useParams only get params from a dynamic route, like` /admin/dev-management/:id`. It can not get params from query URL like `?page=1`. So much to use URLSearchParams to get params.

Reply:
ah ok